### ISSUE TITLE
queue so slow

### ISSUE THREAD

**Issue by reatang** [vladimir-yuldashev/laravel-queue-rabbitmq#30]
It's php-amqplib
![3w ll11rzz cwe8vi9sbd8](https://cloud.githubusercontent.com/assets/8022923/8669310/06e69f9e-2a46-11e5-8837-dcef4df7d018.jpg)
![h9gnklm9yr 9hznqu3 v3h](https://cloud.githubusercontent.com/assets/8022923/8669329/39447934-2a46-11e5-9875-219d58898f9d.jpg)

and

it's laravel-queue-rabbitmq
![lw103b2tc gqt0s7mfkn3u](https://cloud.githubusercontent.com/assets/8022923/8669494/ce3320a8-2a47-11e5-9ada-b875712671c2.jpg)

**Comment by vladimir-yuldashev** [comment]
How do you run your workers? Can you provide more details, please?

**Comment by vladimir-yuldashev** [comment]
Okay, I see that you publish messages instead of consuming. Sorry for previous comment. Attach both of code snippets of how you publish messages.

**Comment by ivan1986** [comment]
```
	public function pushRaw($payload, $queue = null, array $options = [])
	{
		$queue = $this->getQueueName($queue);
		$this->declareQueue($queue);   --- this
		if (isset($options['delay']))
```

**Comment by ivan1986** [comment]
simple fix for 4.2 branch

**Comment by winfan** [comment]
this code in controller 
 for($i=0;$i<=100000;$i++){ 
     $this->dispatch(new SendReminderEmail("hello world rabbitmq".microtime()));
 }

why so slow?
<?php


namespace App\Jobs;

use App\Jobs\Job;
use Illuminate\Contracts\Mail\Mailer;
use Illuminate\Queue\SerializesModels;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Contracts\Bus\SelfHandling;
use Illuminate\Contracts\Queue\ShouldQueue;

class SendReminderEmail extends Job implements SelfHandling, ShouldQueue
{
    use InteractsWithQueue, SerializesModels;

 private $message=null;
    /**
     * Create a new job instance.
     *
     * @return void
     */
    public function __construct($message)
    {
         $this->message = $message;

    }

    /**
     * Execute the job.
     *
     * @return void
     */
    public function handle(Mailer $mailer)
    {
        echo $this->message;
    }
}


Queues

All queues
Filter:	Regex (?)
2 items (show at most )
Overview	Messages	Message rates	+/-
Name
Features	
State
Ready
Unacked
Total
incoming
deliver / get
ack
emails
D	idle	22	0	22	0.00/s		
queue_name
D	running	111,676	0	111,676	0.00/s	5.8/s	5.8/s
